docs/source/basics/chat_datasets.rst

```diff
@@ -0,0 +1,248 @@
+        # By default, user prompt is ignored in loss. Set to True to include it
+        train_on_input=True,
+    )
+    tokenized_dict = ds[0]
+    tokens, labels = tokenized_dict["tokens"], tokenized_dict["labels"]
+    print(m_tokenizer.decode(tokens))
+    # [INST] What is the answer to life?  [/INST] The answer is 42. [INST] That's ridiculous  [/INST] Oh I know.
+    print(labels)
+    # [1, 733, 16289, 28793, 1824, 349, 272, 4372, ...]
+
+.. code-block:: yaml
+
+    # In config
+    tokenizer:
+      _component_: torchtune.models.mistral.mistral_tokenizer
+      path: /tmp/Mistral-7B-v0.1/tokenizer.model
+      prompt_template: torchtune.models.mistral.MistralChatTemplate
+      max_seq_len: 8192
+
+    dataset:
+      _component_: torchtune.datasets.chat_dataset
+      source: json
+      data_files: data/my_data.json
+      split: train
+      conversation_column: conversations
+      conversation_style: sharegpt
+      train_on_input: True
+      new_system_prompt: You are an AI assistant.
+
+Chat dataset format
+-------------------
+
+Chat datasets typically have a single column named "conversations" or "messages" that contains a list of messages on a single topic
+per sample. The list of messages could include a system prompt, multiple turns between user and assistant, and tool calls/returns.
+
+.. code-block:: text
+
+    |  conversations                                               |
+    |--------------------------------------------------------------|
+    | [{"role": "user", "content": "What day is today?"},          |
+    |  {"role": "assistant", "content": "It is Tuesday."}]         |
+    | [{"role": "user", "content": "What about tomorrow?"},        |
+    |  {"role": "assistant", "content": "Tomorrow is Wednesday."}] |
+
+As an example, you can see the schema of the `SlimOrca dataset <https://huggingface.co/datasets/Open-Orca/SlimOrca-Dedup>`_.
+
+Loading chat datasets from Hugging Face
+---------------------------------------
+
+You need to pass in the dataset repo name to ``source``, select one of the conversation styles in ``conversation_style``, and specify the ``conversation_column``.
+For most HF datasets, you will also need to specify the ``split``.
+
+.. code-block:: python
+
+    from torchtune.models.gemma import gemma_tokenizer
+    from torchtune.datasets import chat_dataset
+
+    g_tokenizer = gemma_tokenizer("/tmp/gemma-7b/tokenizer.model")
+    ds = chat_dataset(
+        tokenizer=g_tokenizer,
+        source="Open-Orca/SlimOrca-Dedup",
+        conversation_column="conversations",
+        conversation_style="sharegpt",
+        split="train",
+    )
+
+.. code-block:: yaml
+
+    # Tokenizer is passed into the dataset in the recipe
+    dataset:
+      _component_: torchtune.datasets.chat_dataset
+      source: Open-Orca/SlimOrca-Dedup
+      conversation_column: conversations
+      conversation_style: sharegpt
+      split: train
+
+
+Loading local and remote chat datasets
```

Comment:
is this not already covered above?

Reply:
yeah I guess. was thinking it's worth having a direct section on it, vs just implicit in the example.